logging.js

```diff
@@ -235,11 +238,23 @@ export class LoggingClient {
 	}
 
 	_throttle(log) {
+
+		const now = Date.now();
+
+		// rate limit number of errors to 100 per minute
+		while (this._logHistory.length > 0 && this._logHistory[0] < (now - MAXIMUM_LOGS_TIME_SPAN)) {
+			this._logHistory.shift();
+		}
```

Comment:
Since all of the timestamps would be ordered sequentially, could we just find the right index and slice the array once?

Reply:
Yep -- it would look something like this:

```javascript
let sliceIndex = this._logHistory.length;
for (let i = 0; i < this._logHistory.length; i++) {
	if (this._logHistory[i] >= (now - MAXIMUM_LOGS_TIME_SPAN)) {
		sliceIndex = i;
		break;
	}
}
this._logHistory = this._logHistory.slice(sliceIndex);
```

So 8 lines instead of 3 and IMO much more difficult to wrap my head around what it's trying to do. Not sure if the performance improvement is worth it...

Reply:
Probably not a huge concern since the array is upper bounded by `MAXIMUM_LOGS_PER_TIME_SPAN`